src/pages/understand-stacks/integrate-stacking-delegation.md

```diff
@@ -0,0 +1,196 @@
+```
+
+-> See additional [Stacking library reference](https://github.com/blockstack/stacks.js/tree/master/packages/stacking)
+
+## Step 2: Delegate STX tokens
+
+To get started, delegate STX tokens as an account holder:
+
+```js
+import { getNonce } from '@stacks/transactions';
+import { StacksTestnet, StacksMainnet } from '@stacks/network';
+import { StackingClient } from '@stacks/stacking';
+import BN from 'bn.js';
+
+const network = new StacksTestnet();
+// for mainnet: const network = new StacksMainnet();
+const client = new StackingClient(address, network);
+
+// how much to stack, in microSTX
+const amountMicroStx = new BN(100000000000);
+
+// STX address of the delegator
+const delegateTo = 'ST2MCYPWTFMD2MGR5YY695EJG0G1R4J2BTJPRGM7H';
+
+// burn height at which the delegation relationship should be revoked (optional)
+const untilBurnBlockHeight = 5000;
+
+// a BTC address for reward payouts
+const poxAddress = 'mvuYDknzDtPgGqm2GnbAbmGMLwiyW3AwFP';
+
+// private key of the account holder for transaction signing
+const privateKey = 'd48f215481c16cbe6426f8e557df9b78895661971d71735126545abddcd5377001';
+
+const delegetateResponse = await client.delegateStx({
+  amountMicroStx,
+  delegateTo,
+  untilBurnBlockHeight, // optional
+  poxAddress, // optional
+  privateKey,
+});
+
+// {
+//   txid: '0xf6e9dbf6a26c1b73a14738606cb2232375d1b440246e6bbc14a45b3a66618481',
+// }
+```
+
+This method calls the [`delegate-stx`](/references/stacking-contract#delegate-stx) method of the Stacking contract.
+
+-> To avoid handling private keys, it is recommended to use the [Stacks Wallet](https://www.hiro.so/wallet) to sign the delegation transaction
+
+## Step 3: Stack STX tokens
+
+With an established delegation relationship, the delegator can stack STX tokens on behalf of the account holder.
+
+```js
+// the stacks STX address
+const address = 'ST3XKKN4RPV69NN1PHFDNX3TYKXT7XPC4N8KC1ARH';
+
+// block height at which to stack
+const burnBlockHeight = 2000;
+
+// the delegator initiates a different client
+const delegatorAddress = 'ST22X605P0QX2BJC3NXEENXDPFCNJPHE02DTX5V74';
+
+// number cycles to stack
+const cycles = 3;
+
+// delegator private key for transaction signing
+const delegatorPrivateKey = 'd48f215481c16cbe6426f8e557df9b78895661971d71735126545abddcd5377001';
+
+// the BTC address for reward payouts; either to the delegator or to the BTC address set by the account holder
+const delegatePoxAddress = 'msiYwJCvXEzjgq6hDwD9ueBka6MTfN962Z';
+
+// if you call this method multiple times during the same block, you need to increase the nonce manually
+let nonce = getNonce(delegatorAddress, network);
+nonce = nonce.add(new BN(1));
+
+const delegatorClient = new StackingClient(delegatorAddress, network);
+
+const delegetateStackResponses = await delegatorClient.delegateStackStx({
+  stacker: address,
+  amountMicroStx,
+  poxAddress: delegatePoxAddress, // if set in step 2, this needs to match `poxAddress`
```

Comment:
This must be the BTC address of the delegator. It doesn't work if each stacker uses a different `poxAddress` with the same delegator.

Reply:
why? if the delegation relationship is set with a `pox-addr`, this field needs to match that address: https://github.com/blockstack/stacks-blockchain/blob/master/src/chainstate/stacks/boot/pox.clar#L588.

As far as I understand the same delegator can have stackers with different poxAddr - and would need to make sure to use their btc address when calling this method.

Reply:
If I'm reading that code right, it's a check that "when the delegator calls delegate-stacks-stx, the BTC address they set for PoX rewards _must_ match the approved BTC address that the stacker specified (which is optional)". 

It wouldn't work if the stacker can just set their own BTC address, otherwise we'd break the whole reason for delegation, which is that there are not enough reward slots for everyone to get direct BTC rewards.

Reply:
I'll see what I can do to convey the points in the tutorial. thanks @hstove 🙏🏼